Sources/WebSocketKit/WebSocket.swift

```diff
@@ -164,21 +220,38 @@ public final class WebSocket: Sendable {
         fin: Bool = true,
         promise: EventLoopPromise<Void>? = nil
     ) {
-        let frame = WebSocketFrame(
-            fin: fin,
-            opcode: opcode,
-            maskKey: self.makeMaskKey(),
-            data: data
-        )
-        self.channel.writeAndFlush(frame, promise: promise)
+        if let p = pmce,
+               p.enabled {
+            do {
+                let compressedFrame = try p.compressed(data, fin:fin, opCode: opcode)
+                self.channel.writeAndFlush(compressedFrame, promise: promise)
+            }
+            catch {
+                logger.error("\(error)")
+            }
```

Comment:
Absolutely not.

Reply:
WebSocket.send doesnt throw so this has to stay I think. Dont want to change the signature on send as that would likely break things for people.

Reply:
You can fail the promise (if it's non-nil) on error, e.g. `promise?.fail(error)`

Reply:
removed the log added failed promise.